django/http/request.py

```diff
@@ -347,12 +417,12 @@ def __init__(self, query_string=None, mutable=False, encoding=None):
                 except UnicodeDecodeError:
                     # ... but some user agents are misbehaving :-(
                     query_string = query_string.decode('iso-8859-1')
-            for key, value in parse_qsl(query_string or '',
+            for key, value in parse_qsl(SplitWrapper(query_string or '', limit_checker),
```

Comment:
That's true. We could include parse_qsl() in the Django code but I feel that's adding even more entropy to an already troublesome patch. I don't know how we could enforce the limit in a single pass over the query data in another way.